packages/flutter_tools/lib/src/flutter_manifest.dart

```diff
@@ -0,0 +1,174 @@
+
+  bool get usesMaterialDesign {
+    return _flutterDescriptor.containsKey('uses-material-design') &&
+        _flutterDescriptor['uses-material-design'];
+  }
+
+  List<Map<String, dynamic>> get fontsDescriptor {
+    if (!_flutterDescriptor.containsKey('fonts')) {
+      return <Map<String, dynamic>>[];
+    }
+    return _flutterDescriptor['fonts'];
+  }
+
+  List<String> get assets {
+    if (!_flutterDescriptor.containsKey('assets')) {
+      return <String>[];
+    }
+    return _flutterDescriptor['assets'];
+  }
+
+  List<Font> _fonts;
+
+  List<Font> get fonts {
+    if (!_flutterDescriptor.containsKey('fonts')) {
+      return <Font>[];
+    }
+    if (_fonts == null) {
+      final List<Font> fonts = <Font>[];
+      for (Map<String, dynamic> fontFamily in _flutterDescriptor['fonts']) {
+        final List<Map<String, dynamic>> fontFiles = fontFamily['fonts'];
+        final String familyName = fontFamily['family'];
+        if (familyName == null) {
+          printError('Warning: Missing family name for font.', emphasis: true);
+          continue;
+        }
+        if (fontFiles == null) {
+          printError('Warning: No fonts specified for font $familyName', emphasis: true);
+          continue;
+        }
+
+        final List<FontAsset> fontAssets = <FontAsset>[];
+        for (Map<String, dynamic> fontFile in fontFiles) {
+          final String asset = fontFile['asset'];
+          if (asset == null) {
+            printError('Warning: Missing asset in fonts for $familyName', emphasis: true);
+            continue;
+          }
+
+          fontAssets.add(new FontAsset(
+            asset,
+            weight: fontFile['weight'],
+            style: fontFile['style'],
+          ));
+        }
+        if (fontAssets.isNotEmpty)
+          fonts.add(new Font(fontFamily['family'], fontAssets));
+      }
+
+     _fonts = fonts;
+    }
+
+    return _fonts;
+  }
+
+  static Future<FlutterManifest> createManifestFromPath(String path) async {
+    return  _createManifestFromYaml(_loadFlutterManifest(path));
+  }
+
+  static Future<FlutterManifest> createManifestFromString(String manifest) async {
+    return _createManifestFromYaml(loadYaml(manifest));
+  }
+
+  static  Future<FlutterManifest> _createManifestFromYaml(Object yamlDocument) async {
+    final FlutterManifest pubspec = new FlutterManifest._();
+    if (yamlDocument == null || !await _validate(yamlDocument)) {
+      return null;
+    }
+    pubspec._descriptor = yamlDocument;
+    pubspec._flutterDescriptor =
+        pubspec._descriptor['flutter'] ?? <String, dynamic>{};
+    return pubspec;
+  }
+}
+
+ class Font {
+  final String familyName;
+  final List<FontAsset> fontAssets;
+  Font(
+    this.familyName,
+    this.fontAssets,
+  ) {
+    assert(familyName != null);
+    assert(fontAssets != null);
+    assert(fontAssets.isNotEmpty);
+  }
+
+  @override
+  String toString() => 'Font (family: $familyName, assets: $fontAssets)';
```

Comment:
I believe the idiom is `=> '$runtimeType(x: $_x, y: $_y)'` with no space after the type name. Using `$runtimeType` you don't have to remember to update `toString` if you rename the type.